lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -642,6 +642,31 @@ sub _add_distri_and_version_to_summary {
     }
 }
 
+sub latest_jobs {
+    my ($self, $search_args, $until) = @_;
+    my @latest_jobs;
+
+    my $group_ids = $search_args->{groupids};
+    if (defined $group_ids && @$group_ids > 1) {
+        my @composed_search_arch = map {
+            my $local_search_args = {
+                group_id => $_,
+                build    => $self->schema->resultset('Jobs')->latest_build((groupid => $_))};
+            $local_search_args->{distri}  = $search_args->{distri}  if (defined($search_args->{distri}));
+            $local_search_args->{version} = $search_args->{version} if (defined($search_args->{version}));
+            $local_search_args;
+        } @$group_ids;
+
+        @latest_jobs
+          = $self->schema->resultset('Jobs')->search(\@composed_search_arch)->latest_jobs($until, 1);
+    }
+    else {
+        @latest_jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args)->latest_jobs($until);
+    }
+
+    @latest_jobs;
```

Comment:
Although it is not required in Perl I'd prefer to add a `return` here. And more importantly: Perl will do a copy of the array elements here (and I believe copy elision isn't really a thing in Perl) so please return a reference here (`return \@latest_jobs`). The calling function needs to expect a scalar then (and can just pass it as-is).

Reply:
OTOH it's consistent with the `latest_jobs` method from the schema.
Maybe that one should have been returning an array ref in the first place.